### DISCUSSION TITLE
Log Format Help

### DISCUSSION THREAD

**Discussion by sanind** [sevdokimov/log-viewer#135] [Q&A] [answered]
Python Script logs the data and format is  %(asctime)-15s::%(levelname)s::%(message)s

Below is sample output

2022-08-14 00:40:10,116::INFO::Today is Fun

I tried the below format in conf file but not working, it throws an exception

//    format: {
//      type: Python
//      pattern: "%date{yyyy-MM-dd HH:mm:ss,SSS}::%level::%msg" }

Can pls suggest the right pattern?

**Comment by sevdokimov** [comment, marked as answer]
There is no "Python" log type, try the following:
```
format: {
 type: Log4jLogFormat
 pattern: "%date{yyyy-MM-dd HH:mm:ss,SSS}::%level::%msg" 
}
```

**Comment by sanind** [comment]
@sevdokimov  Thank you it worked

**Comment by kknishev** [comment]
Hi Sergey!
Could you please assist with this timestamp format:
`Jul 17 14:30:49 s9510-order-v2-5fb8bc57c5-b9j4p ws_orderv2[21]`

```
logs = [
  {  path: "**/*.log"
    format: {
type: Log4jLogFormat
pattern: "%date{MMM dd HH:mm:ss} %msg"
// pattern: "%date{MMM dd hh:mm:ss} %msg"
// pattern: "%date{MMM d HH:mm:ss} %msg"
// pattern: "%date{MMM d hh:mm:ss} %msg"
}}]
```
all options are not working with highlighting

**Reply by sevdokimov** [reply]
`pattern: "%date{MMM dd HH:mm:ss} %msg"` must work. It works on my local machine. 
Probably, the language in your OS is not English, in this case java doesn't parse "Jul". You can specify the locale manually:
```
    format: {
      type: Log4jLogFormat
      locale: en_US
      pattern: "%date{MMM dd HH:mm:ss} %msg"
    }
```

**Reply by kknishev** [reply]
perfectly, now working, thanks a lot!!

**Comment by kknishev** [comment]
another call - logs has a stupid double space before DAY and `%date{MMM dd HH:mm:ss}` not working at this case.
Is any way to use two patterns, if first one not working, then using next, or something else?
```
Jun  5 00:00:03 reports01 ucs_point@1[25354]                        <- NOK
Jul 17 14:30:49 s9510-order-v2-5fb8bc57c5-b9j4p ws_orderv2[21]      <- OK
```

btw `%date{MMM d HH:mm:ss}` not working also, only this good  `%date{MMM  d HH:mm:ss}` but in first case only

**Reply by sevdokimov** [reply]
You can use this: `pattern: "%date{MMM [ ]d HH:mm:ss} %msg"`   the space inside '[' ']' is optional.

**Reply by kknishev** [reply]
thanks again, working excellent
